extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/KafkaConnectSource.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.kafka.connect.impl;
+
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import org.apache.kafka.connect.connector.ConnectorContext;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.apache.kafka.connect.source.SourceTaskContext;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.hazelcast.client.impl.protocol.util.PropertiesUtil.toMap;
+import static com.hazelcast.internal.util.Preconditions.checkRequiredProperty;
+import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+
+public class KafkaConnectSource {
+
+    private final SourceConnector connector;
+    private final SourceTask task;
+    private final Map<String, String> taskConfig;
+    /**
+     * Key represents the partition which the record originated from. Value
+     * represents the offset within that partition. Kafka Connect represents
+     * the partition and offset as arbitrary values so that is why it is
+     * stored as map.
+     * See {@link SourceRecord} for more information regarding the format.
+     */
+    private Map<Map<String, ?>, Map<String, ?>> partitionsToOffset = new HashMap<>();
+    private boolean taskInitialized;
+
+    public KafkaConnectSource(Properties properties) {
+        try {
+            String connectorClazz = checkRequiredProperty(properties, "connector.class");
+            Class<?> connectorClass = Thread.currentThread().getContextClassLoader().loadClass(connectorClazz);
```

Comment:
I think it may be a common mistake to not include jar with connector. Maybe you can wrap this in try-catch and provide meaningful context for user?

Reply:
good idea, added: https://github.com/hazelcast/hazelcast/pull/23312/commits/d9e0569b9c1a8f5bd8d9db2ebc8435bc3dfd2f23